jax/_src/lax/control_flow/loops.py

```diff
@@ -2157,8 +2157,46 @@ def fori_loop(lower, upper, body_fun, init_val):
 
 ### map and miscellaneous rules
 
+def _transpose_with_axes(x, axes, *, move_front: bool):
+  flat_axes: list[int] = broadcast_prefix(axes, x, is_leaf=lambda x: x is None)
+  leaves, treedef = tree_flatten(x)
+
+  if move_front:
+    move_axes = lambda leaf, axis: jax.numpy.moveaxis(leaf, axis, 0)
+  else:
+    move_axes = lambda leaf, axis: jax.numpy.moveaxis(leaf, 0, axis)
+
+  leaves = [
+    move_axes(leaf, axis)
+    for leaf, axis in zip(leaves, flat_axes)
+  ]
+  return treedef.unflatten(leaves)
+
+def _batch_and_remainder(x, batch_size: int):
```

Comment:
Can you reuse the utilities `_partition_leading` and `_combine_leading` that are used in the scan unrolling implementation?

Reply:
These functions appear to be deleted after a synced with `main`. Keeping current implementation.